ROOTFS/usr/bin/waggle-network-watchdog

```diff
@@ -79,49 +100,58 @@ def reboot_os():
     subprocess.run(["systemctl", "--force", "reboot"])
 
 
-config = must_read_config_section("/etc/waggle/config.ini", "watchdog")
-ssh_ok_file = config.get("ssh_ok_file", None)
-
-last_connection_time = time_now()
-
 # Recovery actions table [time (s), recovery function]
 # restart networking stack after 15, 20 and 25 of no beehive connectivity
 # reboot after 30 mins of no beehive connectivity
-recovery_actions = [
+#
+# NOTE We sort in increasing order of threshold so that our linear
+# search finds the "earliest" available action
+recovery_actions = sorted([
     (1800, reboot_os),
     (1500, restart_network_services),
     (1200, restart_network_services),
     (900, restart_network_services),
-]
+])
 
-# sort in increasing order of threshold so that our linear
-# search finds the "earliest" available action
-recovery_actions.sort()
-available_actions = recovery_actions.copy()
 
-while True:
-    update_systemd_watchdog()
+def main():
+    logging.basicConfig(level=logging.INFO)
+
+    config = read_watchdog_config("/etc/waggle/config.ini")
 
-    if ssh_connection_ok():
-        logging.info("connection ok")
+    last_connection_time = time_now()
+    available_actions = recovery_actions.copy()
 
-        if ssh_ok_file is not None:
-            Path(ssh_ok_file).touch()
+    while True:
+        update_systemd_watchdog()
+
+        logging.info("checking connection")
+        if require_successive_passes(ssh_connection_ok,
+                                     config.check_successive_passes,
+                                     config.check_successive_seconds):
+            logging.info("connection ok")
+
+            if config.ssh_ok_file is not None:
+                Path(config.ssh_ok_file).touch()
+            else:
+                logging.info("not setting flag for wagman-watchdog")
+            
+            last_connection_time = time_now()
+            available_actions = recovery_actions.copy()
         else:
-            logging.info("not setting flag for wagman-watchdog")
-        
-        last_connection_time = time_now()
-        available_actions = recovery_actions.copy()
-    else:
-        logging.warning(
-            "no connection for %ss", int(seconds_since(last_connection_time))
-        )
+            logging.warning(
+                "no connection for %ss", int(seconds_since(last_connection_time))
+            )
+
+        for thresh, action in available_actions:
+            if seconds_since(last_connection_time) >= thresh:
+                logging.warning("executing %ds recovery action", thresh)
+                action()
+                available_actions.remove((thresh, action))
+                break
+
+        time.sleep(15)
```

Comment:
while you are at it, can you have this be a config too (defaulted to 15s)

Reply:
Sure, just added support for a `check_seconds` config param.